internal/cli/federatedauthentication/federationsettings/connectedorgsconfigs/connect.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package connectedorgsconfigs
+
+import (
+	"context"
+
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/usage"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231115012/admin"
+)
+
+type ConnectOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	cli.InputOpts
+	*DescribeOrgConfigsOpts
+	identityProviderID   string
+	protocol             string
+	federationSettingsID string
+	store                store.ConnectedOrgConfigsUpdater
+}
+
+const (
+	oidc = "OIDC"
+	saml = "SAML"
+)
+
+const updateTemplate = `Connected Org Config connected with {{- range valueOrEmptySlice .DataAccessIdentityProviderIds}}{{.}}{{- end}}.`
+
+func (opts *ConnectOpts) InitStore(ctx context.Context) func() error {
+	return func() error {
+		if opts.store != nil && opts.describeStore != nil {
+			return nil
+		}
+
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		if err != nil {
+			return err
+		}
+
+		return opts.InitDescribeStore(ctx)()
+	}
+}
+
+func (opts *ConnectOpts) Run() error {
+	var orgConfig *atlasv2.ConnectedOrgConfig
+	var err error
+	if orgConfig, err = opts.GetConnectedOrgConfig(opts.federationSettingsID, opts.ConfigOrgID()); err != nil {
```

Comment:
we do a describe first because this API is full of side-effects so better to wrap the request with the exact same config